group_utils/javascript/automation/lib/javascript/community/groupUtils.js

```diff
@@ -0,0 +1,347 @@
+   *
+   * @param {string} group Name of the openHAB group
+   * @returns {Array} Names of all members
+   */
+  context.getAllMembersNames = function (group) {
+    return getAllMembers(group, 'name')
+  }
+
+  /**
+   * Get all (also childs) members' states of a group.
+   *
+   * @param {string} group Name of the openHAB group
+   * @returns {Array} States of all members
+   */
+  context.getAllMembersStates = function (group) {
+    return getAllMembers(group, 'state')
+  }
+
+  /**
+   * Get all (also childs) members' labels of a group.
+   *
+   * @param {string} group Name of the openHAB group
+   * @returns {Array} Labels of all members
+   */
+  context.getAllMembersLabels = function (group) {
+    return getAllMembers(group, 'label')
+  }
+
+  /**
+   * Get all (also childs) members' labels of a group as a concatenated string.
+   *
+   * @param {string} group Name of the openHAB group
+   * @returns {string} Concatenated labels of direct members
+   */
+  context.getAllMembersLabelsString = function (group) {
+    logger.debug('Getting all members of group ' + group)
+    return context.itemRegistry.getItem(group)
+      .getAllMembers()
+      .stream()
+      .map(function (i) {
+        return i.getLabel()
+      })
+      .collect(Collectors.joining(', '))
+  }
+
+  /*
+  Math/arithmetic operations
+  */
+
+  /**
+   * Private function to get numeric states of direct members of a group.
+   * This function only uses Items of type: Number, Dimmer, Rollershutter.
+   * Units of measurement are ignored.
+   *
+   * @param {string} group  Name of the openHAB group
+   * @returns {*} Statistics for Java Collectors.summarizingDouble()
+   */
+  function getMembersNumeric (group) {
+    return context.itemRegistry.getItem(group)
+      .getMembers()
+      .stream()
+      .filter(function (i) {
+        // Log: Check for Item of type: Number, Dimmer or Rollershutter
+        if (!((i.getType() === 'Number') || (i.getType() === 'Dimmer') || (i.getType() === 'Rollershutter'))) {
+          logger.debug(i.getName() + ' ignored, no supported type: ' + i.getType())
+        }
+        // Log: Check for UnDefType (NULL or UNDEF)
+        if (items[i.getName()].class === UnDefType.class) {
+          logger.debug(i.getName() + ' ignored, state is UNDEF or NULL.')
+        }
+        return (
+          // Check for Item of type: Number, Dimmer or Rollershutter
+          ((i.getType() === 'Number') || (i.getType() === 'Dimmer') || (i.getType() === 'Rollershutter')) &&
+          // Check for UnDefType (NULL or UNDEF)
+          (!(items[i.getName()].class === UnDefType.class))
+        )
+      })
+      .collect(Collectors.summarizingDouble(function (i) {
+        return parseFloat(i.getState())
+      }))
+  }
+
+  /**
+   * Get the sum of direct members' number states.
+   *
+   * @param {string} group Name of the openHAB group
+   * @returns {number} Sum of direct members number states
+   */
+  context.MembersSum = function (group) {
```

Comment:
Typical naming conventions have only Class names starting with capital letters and then using camel case. Variables and methods/function start with lower case and use camel case. Constants use all caps and underscores.